src/main/java/com/api/controller/RocketController.java

```diff
@@ -15,9 +19,15 @@
 public class RocketController {
 
     private final RocketPresenter rocketPresenter;
+    private final RocketPersistenceService rocketPersistenceService;
+    private final RocketService rocketService;
```

Comment:
please separate here the services, controller should be aware only of 1 service or presenter, you should have Controler -> Presenter-> Presenter is responsible of doing the logic, so he is aware of Service, and Persistence, Fetchers etc. , same in the presenter we are making the mapping onto the DTO and sending the response back to the Controller